compiler/CodeGenVisitor.cpp

```diff
@@ -42,52 +44,99 @@ antlrcpp::Any CodeGenVisitor::visitInit(ifccParser::InitContext *ctx)
 	string varname = ctx->VARNAME()->getText();
 	string value = visit(ctx->expression()).as<string>();
 	int index = (this->vars.size() + 1) * 8;
-
-	if( vars.find(varname)==vars.end()){
-		this->vars[varname]=index;
-		cout << "\tmovl " + value + ", -" + to_string(index) + "(%rbp)" << endl;
-	  return 0;
-	}else{
-		error=true;
-		return 0;
+	if (vars.find(varname) == vars.end()) {
+		this->vars[varname] = index;
+		cout << "\tmovl " + value + ", " << EAX << endl;
+	  cout << "\tmovl " + EAX + ", -" + to_string(index) + "(%rbp)" << endl;
+	} else {
+		error = true;
 	}
-	
-	
-
-	
-
+	return 0;
 }
 
 antlrcpp::Any CodeGenVisitor::visitValue(ifccParser::ValueContext *ctx) 
 {
 	string returnval;
 	antlr4::tree::TerminalNode * varnameNode = ctx->VARNAME();
 	if (varnameNode) {
-		std::string varname = varnameNode->getText();
-		std::string index = std::to_string(this->vars[varname]);
+		string varname = varnameNode->getText();
+		string index = to_string(this->vars[varname]);
 		returnval = "-" + index + "(%rbp)";
 	} else {
 		returnval = "$" + ctx->CONST()->getText();
 	}
 	return returnval;
 }
 
+string CodeGenVisitor::operationExpression(string leftval, string rightval, string operation) {
```

Comment:
It would be nice to have some comments for others to understand what is going on here without having to ask you